src/functions/anti-scam/checkScam.ts

```diff
@@ -8,26 +9,25 @@ import { resolveRedirect } from '../../util/resolveRedirect';
 export async function checkScam(content: string): Promise<string[]> {
 	const redis = container.resolve<Redis>(kRedis);
 
-	const urlRegex =
-		/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/gi;
+	const linkRegex = /(?:https?:\/\/)(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b[-a-zA-Z0-9@:%_\+.~#?&//=]*/gi;
 	const scamDomains = await redis.smembers('scamdomains');
 	const trippedDomains = [];
 
 	let matches: any[] | null = [];
-	while ((matches = urlRegex.exec(content)) !== null) {
-		const url = matches[0];
-		const urlLower = url.toLowerCase();
-		const hit = scamDomains.find((d) => urlLower.includes(d.toLowerCase()));
+	while ((matches = linkRegex.exec(content)) !== null) {
+		const url = new URL(matches[0]);
```

Comment:
If I send `https://::.com`, which the RegExp matches, this call which throw a `ReferenceError` exception.

Probably make a tiny function wrapping `new URL(...)` inside a `try/catch` returning `null` if the parse fails?